### ISSUE TITLE
Uncaught Error: toAST() expects a succesfull MatchResult as first parameter, but the MatchResult is successful

### ISSUE THREAD

**Issue by nailuj29** [harc/ohm#363]
I am creating an app that uses Ohm, and when trying to get an AST with the `toAST` function. However, I get an error stating `toAST() expects a succesfull MatchResult as first parameter, but the MatchResult is successful` I believe this is an error with the `instanceof` in the code, because `semanticsForToAST` fails similarly, and the `MatchResult` is successful

**Comment by pdubroy** [comment]
I wasn't able to reproduce this. Can you give me some more details about how are importing Ohm (e.g., `require` or `import`), and how you are running and/or bundling it?

It sounds like the problem is that there are two different instances of the module.

**Comment by nailuj29** [comment]
I am importing it using `import` and running it in a Vite with typescript.

**Comment by pdubroy** [comment]
@nailuj29 Can you show me exactly how you are importing and using it? If you give me the code it's much easier for me to try to reproduce.

Also, what does your tsconfig look like? That could make a difference.

**Comment by pdubroy** [comment]
Ok, I was able to reproduce the problem as follows:

1. Set up a new Vite project with `npm create vite@latest`, and chose "vanilla-ts" as the project type.
2. `npm install -D ohm-js`
3. Put the following code into main.ts:
   ```
   import ohm from 'ohm-js';
   import extras from 'ohm-js/extras';

   extras.toAST(ohm.grammar('G { start = }').match(''))
   ```
@nailuj29, is this similar to how you are using it?

**Comment by nailuj29** [comment]
Yes, that is what I am doing

**Comment by pdubroy** [comment]
@nailuj29 The issue is fixed now, and I've just published the fix in [v16.3.1](https://www.npmjs.com/package/ohm-js/v/16.3.1).
